lib/puppet/provider/service/service.rb

```diff
@@ -36,6 +36,9 @@ def ucommand(type, fof = true)
     else
       cmd = [send("#{type}cmd")].flatten
     end
+    if (( type =~ /^start/ ) && ( Facter.value(:osfamily) == "Solaris" ))
+      cmd = ["/usr/bin/ctrun -l none", cmd].flatten.join(" ")
+    end
```

Comment:
It feels wrong to put this logic in the base `service` provider. Since we already have an `smf` service provider, how about if `smf` just overrides the `texecute` method:

```
def texecute(type, command, fof = true, squelch = false, combine = true)
  if type =~ /^start/
    command =  ["/usr/bin/ctrun -l none", command].flatten.join(" ")
  end
  super(type, command, fof, squelch, combine)
end
```